Add "No data" Guidance for NameValueList/Empty DataTable Cell (#2269)
grommet/hpe-design-system: aries-site/src/pages/components/namevaluelist.mdx

  <NameValueListReadMoreExample />
</Example>

## Name-Value Pair with Empty Value

> If the value of a name-value pair is empty, use `--` in place of where a value would appear. This visual indicator aids the user in identifying fields that do not have values without increasing cognitive load.

Reviewer: ```suggestion
If the value of a name-value pair is empty, use `--` in place of where a value would appear. This visual indicator aids the user in identifying fields that do not have values, without increasing cognitive load.
```

Reviewer: I'd maybe add a comma here to clarify the separation between 'values' and 'without'.

> Add an `a11yTitle="No value"` to the Text component that would otherwise render the data to ensure users of assisstive technology are informed of the empty state.

Reviewer: ```suggestion
To inform assistive-technology users of an empty value, add an `a11yTitle="No value"` to the Text component that would otherwise render the data.
```
I'm always torn between presenting the action first or the condition in which it's used but I default to the latter for slightly more complicated sentences. The original still works though so this is 100% your call.

Reviewer 2: ☝️ I prefer @CitizenFox's approach. I too like to follow the format which leads with "what is being accomplished" followed by "how to achieve."

  <NameValueListReadMoreExample />
</Example>

## Name-Value Pair with Empty Value

> If the value of a name-value pair is empty, use `--` in place of where a value would appear. This visual indicator aids the user in identifying fields that do not have values without increasing cognitive load.

Reviewer 2: Thinking this would treat the accessibility aspect as more of a first-class consideration.

```suggestion
If the value of a name-value pair is empty, use `--` in place of where a value would appear. This visual indicator aids the user in identifying fields that do not have values without increasing cognitive load.

Add an `a11yTitle="No data"` to the Text component that would otherwise render the data to ensure users of assistive technology are informed of the empty state.
```

Author: Done.

  height={{ min: 'small' }}
  pad="large"
  showResponsiveControls={false}
  template
>
  <NameValueListEditExample />
</Example>

**Summaries (such as Dashboard Cards)** - Summaries are intended to display information-salient content, often in areas with restricted screen real estate. Given this, name-value pairs without a value should be omitted.

<Example
  docs="https://v2.grommet.io/namevaluelist?theme=hpe#layout"
  code={["https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/namevaluelist/Examples/NameValueListSummaryExample.js",
    "https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/namevaluelist/data.js",
  ]}
  github="https://github.com/grommet/hpe-design-system/tree/master/aries-site/src/examples/components/namevaluelist/Examples/NameValueListSummaryExample.js"
  height={{ min: 'small' }}
  pad="large"
  showResponsiveControls={false}
  template
>
  <NameValueListSummaryExample />
</Example>

### Accessibility for Name-Value Pairs with Empty Value

> Add an `a11yTitle="No data"` to the Text component that would otherwise render the data to ensure users of assisstive technology are informed of the empty state.

Reviewer 2: This feels a bit buried to me. I wonder if we should pull this up to L213. Making a comment/suggestion there.
